### PR TITLE
SPIFFE trust bundle endpoint

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

Adds [SPIFFE trust bundle endpoint](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE_Trust_Domain_and_Bundle.md#41-jwk-set) support.

Trust bundle CA certs can be specified as K8s secrets with type `istio.banzaicloud.io/trust-bundle-ca` and specific labels that connects it to a control plane. Trust bundles are exposed on the SPIFFE endpoint (`/spiffe-trust-bundle?trustDomain=<trust-domain>&revision=<control-plane-revision>`) of the webhook server.

The following secret specifies a trusted CA cert for `somecluster.somewhere.at.some.corp` trust domain and for `cp-v18x.istio-system` Istio control plane:

```
apiVersion: v1
data:
  certs: <base64 encoded CA cert>
kind: Secret
metadata:
  labels:
    istio.banzaicloud.io/trust-domain: somecluster.somewhere.at.some.corp
    istio.io/rev: cp-v18x.istio-system
  name: somecluster-ca
  namespace: istio-system
type: istio.banzaicloud.io/trust-bundle-ca
```

### Why?

It is a building block to provide easy SPIFFE federation to seamlessly connect separate meshes with [Backyards](https://banzaicloud.com/products/backyards/).

### Checklist

- [x] Implementation tested

### CODE REVIEW

File: pkg/resources/webhookcert/configmap.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+func GetWebhookCABundles(client client.Client, webhookConfigurationName string) (string, error) {
+	bundles := make([]string, 0)
+
+	vwh := &admissionregistrationv1.ValidatingWebhookConfiguration{}
+	err := client.Get(context.Background(), types.NamespacedName{
+		Name: webhookConfigurationName,
+	}, vwh)
+	if err != nil {
+		return "", err
```
Comment (Laci21) [inline pkg/resources/webhookcert/configmap.go:60]:
```suggestion
		return "could not get validating webhook", err
```

File: pkg/trustbundle/secret.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 Banzai Cloud.
```
Comment (Laci21) [inline pkg/trustbundle/secret.go:2]:
```suggestion
Copyright 2021 Banzai Cloud.
```

File: pkg/trustbundle/trustbundle.go [hunk 3]
```diff
@@ -0,0 +1,282 @@
+/*
+Copyright 2019 Banzai Cloud.
```
Comment (Laci21) [inline pkg/trustbundle/trustbundle.go:2]:
```suggestion
Copyright 2021 Banzai Cloud.
```

File: pkg/trustbundle/trustbundle.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	istiov1beta1 "github.com/banzaicloud/istio-operator/pkg/apis/istio/v1beta1"
+)
+
+const (
+	CertsDataKey                              = "certs"
+	RevisionLabel                             = "istio.io/rev"
```
Comment (Laci21) [inline pkg/trustbundle/trustbundle.go:40]: Isn't this const defined somewhere else in this repo? If so, could you reuse that?

**Review by Laci21 — APPROVED** [review]
Awesome, thank you very much @waynz0r, only a few nits!

**Review by khernyo — APPROVED** [review]

**Pull request merged**